diff --git a/CMakeLists.txt b/CMakeLists.txt
index c7193fbb..bcd14837 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -152,8 +152,7 @@ add_subdirectory("common")
 add_subdirectory("${_OSNAME}")
 add_library(libui ${_LIBUI_SOURCES})
 target_include_directories(libui
-	PUBLIC .
-	PRIVATE ${_LIBUI_INCLUEDIRS})
+    PUBLIC $<INSTALL_INTERFACE:include>)
 target_compile_definitions(libui
 	PRIVATE ${_LIBUI_DEFS})
 # cmake produces this for us by default but only for shared libraries
@@ -229,10 +228,30 @@ macro(_add_exec _name)
 	# TODOfor some reason these don't propagate
 	if(NOT WIN32)
 		target_include_directories(${_name}
-			PUBLIC .)
+			PUBLIC $<INSTALL_INTERFACE:include>)
 		target_compile_options(${_name}
 			PUBLIC ${_COMMON_CFLAGS})
 	endif()
 endmacro()
 add_subdirectory("test")
 add_subdirectory("examples")
+
+if(BUILD_SHARED_LIBS)
+    install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}-config
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+else()
+    install(TARGETS ${PROJECT_NAME}
+        EXPORT ${PROJECT_NAME}-config
+        DESTINATION ${CMAKE_INSTALL_LIBDIR})
+endif()
+
+install(EXPORT ${PROJECT_NAME}-config
+    FILE ${PROJECT_NAME}-config.cmake
+    NAMESPACE libui::
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
+)
+
+install(FILES ui.h DESTINATION include)
diff --git a/unix/CMakeLists.txt b/unix/CMakeLists.txt
index 5c9425d2..de1cb2d2 100644
--- a/unix/CMakeLists.txt
+++ b/unix/CMakeLists.txt
@@ -1,8 +1,19 @@
 # 3 june 2016
 
 find_package(PkgConfig REQUIRED)
+
+if (NOT PKG_CONFIG_FOUND)
+    message(FATAL_ERROR "pkg-config is required.")
+endif ()
+
 pkg_check_modules(GTK REQUIRED gtk+-3.0)
 
+if (NOT GTK_FOUND)
+    message(FATAL_ERROR "GTK+ 3.0 is required.")
+else ()
+    add_definitions(${GTK_CFLAGS} ${GTK_CFLAGS_OTHER})
+endif ()
+
 list(APPEND _LIBUI_SOURCES
 	unix/alloc.c
 	unix/area.c
@@ -62,5 +73,5 @@ set(_LIBUI_CFLAGS
 PARENT_SCOPE)
 
 set(_LIBUI_LIBS
-	${GTK_LDFLAGS} m ${CMAKE_DL_LIBS}
+	${GTK_LIBRARIES} m ${CMAKE_DL_LIBS}
 PARENT_SCOPE)
